### PR TITLE
[release-1.24] Fix GitHub actions CI

### PR DESCRIPTION
**Pull request by jrvaldes**
This is a manual cherry-pick of https://github.com/cri-o/cri-o/pull/6505

```release-note
None
```

### CODE REVIEW

File: internal/factory/container/container.go [hunk 1]
```diff
@@ -144,6 +144,8 @@
 }
 
 // SpecAddMount adds a specified mount to the spec
+//
+//nolint:gocritic // passing the spec mount around here is intentional
```
[the remark this answers is not in the document]
Comment (jrvaldes) [reply internal/factory/container/container.go:148]: @cri-o/cri-o-maintainers  I may need directions here; Is this required in release-1.24?  `sonatype-lift` bot did not complain in 1.25 or 1.26 PRs.

[the remark this answers is not in the document]
Comment (haircommander) [reply internal/factory/container/container.go:148]: it's possible we'd not enabled the gocritic linter here

[the remark this answers is not in the document]
Comment (jrvaldes) [reply internal/factory/container/container.go]: Removed the unused linter instructions in https://github.com/cri-o/cri-o/pull/6506/commits/484e2678461a97ef742372a7b0741348e5198a8f

**Comment by haircommander** [comment]
/approve

LGTM, thanks!

**Comment by jrvaldes** [comment]
Set release-note as `None` in the PR description as per [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md)

**Pull request merged**